cfspopcon/formulas/inductances.py

```diff
@@ -0,0 +1,626 @@
+        )
+
+    return vertical_magnetic_field
+
+
+### ANALYTIC FUNCTIONS FOR PLASMA EXTERNAL INDUCTANCE AND VERTICAL FIELD ###
+
+
+@wraps_ufunc(
+    input_units=dict(inverse_aspect_ratio=ureg.dimensionless, surface_inductance_coefficients=None),
+    return_units=dict(func_term=ureg.dimensionless),
+    pass_as_kwargs=("surface_inductance_coefficients",),
+)
+def calc_fa_Sums_Na(inverse_aspect_ratio: float, surface_inductance_coefficients: SurfaceInductanceCoeffs) -> Unitfull:
+    """Calculate a sum for eq. 17 on page 6 in...
+
+    A power-balance model for local helicity injection startup in a spherical tokamak :cite:`Barr_2018`
+    NOTE: Default values for for the coefficients 'N[a,d]' and '[a,e]' are taken from `Barr_2018` which are obtained
+    from fitting them to model flux_PF and flux_I obtained from over 330 model equilibria spanning 0<=delta<=0.5 whereas
+    SPARC is projected to have delta95 = 0.54
+
+    Args:
+        inverse_aspect_ratio: [~] :term:`glossary link<inverse_aspect_ratio>`
+        surface_inductance_coefficients: [~] :term:`glossary link<surface_inductance_coefficients>`
+
+    Returns:
+         functional term [~]
+    """
+    Na = 4
+    coeffs = select_coeffs(surface_inductance_coefficients)
+    a = coeffs[0]
+
+    sum1 = 0
+    sum2 = 0
+    n = np.arange(Na // 2)
+    m = n + 1
+    sum1 = np.sum(a[n] * (np.sqrt(inverse_aspect_ratio)) ** m)
+    sum2 = np.sum(a[(Na // 2) + n] * (np.sqrt(inverse_aspect_ratio)) ** m)
+    return float(sum1), float(sum2)
+
+
+@wraps_ufunc(
+    input_units=dict(inverse_aspect_ratio=ureg.dimensionless, surface_inductance_coefficients=None),
+    return_units=dict(func_term=ureg.dimensionless),
+    pass_as_kwargs=("surface_inductance_coefficients",),
+)
+def calc_fa_Sum_Ne(inverse_aspect_ratio: float, surface_inductance_coefficients: SurfaceInductanceCoeffs) -> Unitfull:
+    """Calculate a sum for eq. 17 on page 6 in...
+
+    A power-balance model for local helicity injection startup in a spherical tokamak :cite:`Barr_2018`
+
+    Args:
+        inverse_aspect_ratio: [~] :term:`glossary link<inverse_aspect_ratio>`
+        surface_inductance_coefficients: [~] :term:`glossary link<surface_inductance_coefficients>`
+
+    Returns:
+         functional term [~]
+    """
+    Ne = 4
+    coeffs = select_coeffs(surface_inductance_coefficients)
+    e = coeffs[4]
+
+    n = np.arange(0, Ne)
+    m = n + 1
+    return float(np.sum(e[n] * np.sqrt(inverse_aspect_ratio) ** m))
+
+
+@wraps_ufunc(
+    input_units=dict(inverse_aspect_ratio=ureg.dimensionless, surface_inductance_coefficients=None),
+    return_units=dict(func_term=ureg.dimensionless),
+    pass_as_kwargs=("surface_inductance_coefficients",),
+)
+def calc_fb_Sum_Nb(inverse_aspect_ratio: float, surface_inductance_coefficients: SurfaceInductanceCoeffs) -> Unitfull:
+    """Calculate the sum for eq. 18 on page 6 in..."""
```

Comment:
Is this missing the last part of the ref?

Reply:
Kinda. It is the paper by Barr, but I didn't want to write it again because it is the same for every functional. I can write it out for each real quick if you'd like.

Reply:
Would you like me to add the citation for each?